File: tests/fixtures/resend/resend-missing-idempotency-key-fixed/03_generic_transport_wrapper_adversarial.py
import os
import resend

resend.api_key = os.environ["RESEND_API_KEY"]


def send_email(to: str, subject: str, html: str):
    # Subject and body are both caller-supplied: this function cannot name the
    # operation it performs, so no key is derivable here.
    return resend.Emails.send({"from": "Acme <a@acme.com>", "to": to, "subject": subject, "html": html})

File: tests/fixtures/resend/resend-missing-idempotency-key-fixed/03-generic-transport-wrapper-adversarial.ts
import { Resend } from 'resend';

const resend = new Resend(process.env.RESEND_API_KEY);

// Both subject and body are caller-supplied, so this function cannot know which
// logical operation it is performing and has no name to seed a key with. The
// key belongs at the call sites, where the intent is known.
export async function sendEmail({
  from,
  email,
  subject,
  html,
}: {
  from: string;
  email: string;
  subject: string;
  html: string;
}) {
  return await resend.emails.send({ from, to: email, subject, html });
}

// Same shape with positional parameters and a derived subject.
export async function forward(to: string, subject: string, text: string) {
  return resend.emails.send({
    from: 'Acme <relay@acme.com>',
    to,
    subject: `Fwd: ${subject}`,
    text,
  });
}
